flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/cache/AbstractDownloadCache.java

```diff
@@ -91,6 +91,30 @@ public void afterTestSuccess() {
 
 	abstract boolean matchesCachedFile(Matcher matcher, String url);
 
+	private long getDownloadTimeoutMillis() {
+		String downloadTimeoutProp = System.getProperty("downloadTimeoutMillis");
+		long downloadTimeout = 5 * 60 * 1000; // default 5min.
+		if (downloadTimeoutProp != null && downloadTimeoutProp.length() > 0) {
+			downloadTimeout = Long.parseLong(downloadTimeoutProp);
+			if (downloadTimeout <= 0) {
+				throw new IllegalArgumentException("Illegal property setting: -DdownloadTimeoutMillis=" + downloadTimeoutProp);
+			}
+		}
+		return downloadTimeout;
+	}
+
+	private int getDownloadMaxRetries() {
+		String downloadMaxRetriesProp = System.getProperty("downloadMaxRetries");
```

Comment:
Let's refactor this to be a `ParameterProperty`.

Reply:
Really good advise,   will do the refactor.